File: src/domain/repositories/gym.repository.ts
import { IGymEntity, LocationEntity } from "../entities";
import dataSources from "../../data/data-sources/injection";

class GymRepository {
  create(gym: Omit<IGymEntity, "_id">) {
    return dataSources().gymDatabase.create(gym);
  }

  getNearest(location: LocationEntity, meters: number, amount?: number) {
    return dataSources().gymDatabase.getNearest(location, meters, amount);
  }

  get(gymId: string) {
    return dataSources().gymDatabase.get(gymId);
  }

  deleteAll() {
    return dataSources().gymDatabase.deleteAll();
  }
}

export const gymRepo = new GymRepository();

File: lib/src/domain/repositories/gym.repository.dart
import '../../common/failure.dart';
import '../../common/utils/utils.dart';
import '../../data/data_sources/gym/gym.dart';
import '../entities/entities.dart';

abstract class GymRepository {
  static FEither<List<GymEntity>, Failure> getNearest(
    LocationEntity location,
    double meters, {
    int? amount,
  }) =>
      GymDataSource.getNearest(
        location,
        meters,
        amount: amount,
      );
}
